### DISCUSSION TITLE
question about carbene like molecules

### DISCUSSION THREAD

**Discussion by qitianya** [rdkit/rdkit#7916] [Q&A]
I wonder to konw that N1C=C[N+]=[C]1 is a carbene smiles or just a normal thiazole, aka how to express the implicit hydrogens or lone pairs with smiles?

I'm a freshman, actually i have read the daylight and some discussions, while i still not sure could i use this smiles to substructure search to pure dataset. So forgive me😂

**Comment by qitianya** [comment]
I have tried those smiles, like N1C=C[N+]=[CX2&H0]1, but the code couldn't work.

**Comment by ptosco** [comment]
That is neither a thiazole nor a carbene, that is a protonated imidazole, aka [imidazolium cation](https://pubchem.ncbi.nlm.nih.gov/compound/Imidazolium), with two radical electrons, as per its depiction:
![image](https://github.com/user-attachments/assets/3d200203-bb35-4d41-be36-df420219dd52)
As you may find in the Daylight documentation, hydrogens in SMILES are implicit only when elements are not enclosed between square brackets. When they are enclosed in square brackets, as is the case for elements bearing a formal charge such as the N+ in your SMILES, you need to provide explicit hydrogens between brackets. So the SMILES encoding a stable, non-radical imidazolium cation is `N1C=C[NH+]=C1`, or `[nH]1cc[nH+]c1` in aromatic SMILES notation.

**Reply by qitianya** [reply]
Well, I understand what you mean, and your example has helped me understand a little better.

What I actually wanted to ask about is the SMILES of the N-heterocyclic carbene class of molecules (Figure 1)
<img width="603" alt="截屏2024-10-14 下午8 25 07" src="https://github.com/user-attachments/assets/bbb48b4a-db76-4ff8-bc74-cb4377c47b02">
, but in actual reactions, people mostly feed in the form of the precursor imidazolium (Figure 2)
<img width="598" alt="截屏2024-10-14 下午8 26 09" src="https://github.com/user-attachments/assets/25618beb-9d18-4910-b643-e578532bf29b">.

I mainly use the substructure search function developed based on RDkit (https://chemistry.stackexchange.com/questions/128440/substructure-search-with-rdkit?newreg=3d8a0e98b4ea4ba4a69648fdf00ae5ea) to obtain right data with the structure of Figure 1 and Figure 2 from public databases.

However, due to the characteristics of carbene/imidazolium (radical), I'm not sure if the smiles I wrote for them are correct?
![image](https://github.com/user-attachments/assets/9eddd214-24ac-432e-b1ab-8aa649a068dc)

**Comment by ptosco** [comment]
The correct SMILES for Figure 1 is `[*]N1C=CN(*)[C]1`.
The correct SMILES for Figure 2 is `[*]N1C=C[N+](*)=[C]1`.
Not sure what public databases you wish to search, but please consider most databases will not contain unstable structures such as carbenes and radicals.
